src/plots/barPlot.ts

```diff
@@ -119,14 +127,30 @@ export class Bar<X, Y> extends XYPlot<X, Y> {
       super.y(<number | IAccessor<number>>y);
     } else {
       super.y(<Y | IAccessor<Y>>y, yScale);
-      yScale.onUpdate(this._updateBarPixelWidthCallback);
+      yScale.onUpdate(this._updateBarPixelThicknessCallback);
     }
 
-    this._updateValueScale();
+    this._updateLengthScale();
     return this;
   }
 
   /**
+   * The binding associated with bar length. Length is the count or value the bar is trying to show.
+   * This is the .y() for a vertical plot and .x() for a horizontal plot.
+   */
+  protected length(): Plots.ITransformableAccessorScaleBinding<any, number> {
+    return this._isVertical ? this.y() : this.x();
```

Comment:
length is perpendicular to thickness - it represents the actual quantity the bar is representing (while thickness/position shows the category that the bar is associated with)